design/secret_rotation_design.md

```diff
@@ -0,0 +1,951 @@
+application restart e.g. when Conjur secrets change must be run as the same
+UID as the Secrets Provider. The risk introduced by requiring that all
+containers share a process namespace can be mitigated by running all other
+containers that do not require process restart using a different UID as the
+Secrets Provider. These other processes:
+
+- Can "see" other processes running in other containers in the Pod, BUT...
+- Cannot send or receive signals to/from the Secrets Provider, and...
+- Cannot access the file system of the Secrets Provider or the application
+  container that is running with the same UID as the Secrets Provider.
+
+## Audit
+
+No changes to Conjur audit behavior are required for this feature.
+
+##  Development Tasks
+
+Development tasks for this feature are organized into tasks corresponding to
+three phases of development:
+
+- Minimally-featured community release
+- Full-featured GA release
+
+### Development Tasks: Minimally-Featured Community Release
+
+| Description | Jira Story | Estimated<br />Story<br />Points<br /> | Completed |
+|-------------|------------|-----------|-----------|
+| Solution design approval | [ONYX-15257](https://ca-il-jira.il.cyber-ark.com:8443/secure/RapidBoard.jspa?rapidView=1591&view=detail&selectedIssue=ONYX-15257) | 5 | |
+| Security review approval (minimally-featured Community release) | | 1 | |
+| Annotation Parsing and Validation | | 3 | |
+| Implement periodic provider (without UT) | | 3 | |
+| Periodic provider UT | | 3 | |
+| Detect changes in rendered file content (`file` mode) with UT | | 5 | | 
+| Create a new secrets rotation markdown file (e.g. "SECRET_ROTATION.md") in GitHub repo | | 5 | |
+
+### Development Tasks: Full-Featured GA Release
+
+| Description | Jira Story | Estimated<br />Story<br />Points<br /> | Completed |
+|-------------|------------|-----------|-----------|
+| Implement atomic writes for multiple Conjur secret values | | 5 | |
+| Implement reporting of multiple errored secret variables for bulk Conjur secret retrieval | | 5 | |
+| Implement secret file deletion upon secret deleted / access being revoked | | 3 | |
+| Implement K8s Secret secret values deletion upon secret deleted / access being revoked | | 3 | |
+| Implement atomic write of secret files | | | |
+| Implement detection of changes in K8s Secret content with UT | | 3 | |
+| Implement E2E tests for SP Sidecar (with/without rotation) | | 3 | |
+| Implement process restart with UT | | 3 | | 
+| Measure incremental Push-to-file latency using Secrets Provider tracing | | 1 | |
+| Measure incremental `k8s-secret` mode latency using Secrets Provider tracing | | 1 | |
+| Work with tech writers to create official documentation for secrets rotation | | 2 | |
+| Katacoda/Instruqt Demo ??? | | 3 | |
+| Document process restart in GitHub README | | 2 | |
+| Work with tech writers to create official documentation for process restart | | 1 | |
```

Comment:
If we're only doing a community-level security review before, we'll need a deeper dive here (plus can cover all the additional implement tasks).

Reply:
Added.